### DISCUSSION TITLE
LC0024 vs Find Event

### DISCUSSION THREAD

**Discussion by pri-kise** [StefanMaron/BusinessCentral.LinterCop#352] [General]
I like the idea to avoid an unncessary semicolon.

https://github.com/StefanMaron/BusinessCentral.LinterCop/wiki/LC0024

I have one problem with this rule currently.
Th Find Event Action adds automatically a semicolon.
Therefore I don't know why Microsoft does this, but maybe we'll need an issue on the AL Language Repo to dicuss this code styling warning.

**Comment by fvet** [comment]
When reporting to MS, maybe also include the 'Subscribe to Event' (AL Explorer) and 'Implement interface' code action as reported here
https://github.com/StefanMaron/BusinessCentral.LinterCop/issues/331

**Reply by pri-kise** [reply]
Well at first the documentation for the AL Language should mention this best practice on semicolons.
Maybe anywhere on the statement documentation of the AL Language.
https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/devenv-al-control-statements#al-compound-statements

**Reply by Arthurvdv** [reply]
I've started an topic on Twitter/X to receive some help on this (https://twitter.com/arthrvdv/status/1727729430737342791) and our BC community is awesome! 🥳 Also interesting to read the details behind this semicolon-thing.

A) Created an issue for the AL Language to change the default behavior on this: https://github.com/microsoft/AL/issues/7598.
B) @waldo1001 had voluntarily created an PR to remove this from his snippets: https://github.com/waldo1001/crs-al-language-extension/pull/294.

From a code linting perspective, this is a great step in the right direction! With both feet on the ground: _"..it's just a semicolon.."_ 😅

I'm not sure what to do in de statement documentation of the AL Language. If as default is never is being added, do we need to explicit document this in the documentation that it's not necessary to add something that's default not there?

**Reply by Arthurvdv** [reply]
@pri-kise, great work on https://github.com/MicrosoftDocs/dynamics365smb-devitpro-pb/pull/3127

**Reply by fvet** [reply]
@pri-kise  Did you miss some or was I too eager in fixing some other code samples too?

https://github.com/MicrosoftDocs/dynamics365smb-devitpro-pb/pull/3131/files

**Reply by Arthurvdv** [reply]
I've asked @anzwdev for an code action in the AZ AL Dev Tools here: https://github.com/anzwdev/al-code-outline/issues/510.

This wil be much more easily to use then the current RegEx in the wiki on this rule.
